core/server/update-check.js

```diff
@@ -161,16 +163,16 @@ function updateCheckResponse(response) {
         api.settings.edit(
             {settings: [{key: 'nextUpdateCheck', value: response.next_check}]},
             internal
-        ).catch(errors.rejectError),
+        ).catch(errors.rejectError).reflect(),
         api.settings.edit(
             {settings: [{key: 'displayUpdateNotification', value: response.version}]},
             internal
-        ).catch(errors.rejectError)
+        ).catch(errors.rejectError).reflect()
     );
 
-    return Promise.settle(ops).then(function then(descriptors) {
+    return Promise.map(ops).then(function then(descriptors) {
```

Comment:
Actually I think in the catch we can just need to return the error as a rejected promise again and we can keep the logic the same for now.